test/misc.jl

```diff
@@ -61,3 +61,36 @@ for backend in [
 ]
     println(backend)
 end
+
+using Setfield
+
+@testset "Setfield compatibility" begin
```

Comment:
I haven't encountered Setfield and ConstructionBase before. IIUC these tests would have failed without the extension?

Reply:
Some would have passed of course, but there are a few with required type parameters which would have failed

Reply:
yes